core/bytes/buffer.odin

```diff
@@ -310,6 +310,26 @@ buffer_unread_rune :: proc(b: ^Buffer) -> io.Error {
 	return nil
 }
 
+buffer_seek :: proc(b: ^Buffer, offset: i64, whence: io.Seek_From) -> (i64, io.Error) {
+	abs: i64
+	switch whence {
+	case .Start:
+		abs = offset
+	case .Current:
+		abs = i64(b.off) + offset
+	case .End:
+		abs = i64(len(b.buf)) + offset
+	case:
+		return 0, .Invalid_Whence
+	}
+
+	if abs < 0 {
+		return 0, .Invalid_Offset
+	}
+	b.last_read = .Invalid
+	b.off = int(abs)
```

Comment:
I would do the cast before the check for it being negative so you automatically handle overflows

Reply:
Good catch. However, now I'm wondering if the buffer should just store an `i64` anyway.

Reply:
I think that's fine too

Reply:
I read and thought more about this and decided to go with your original suggestion on the basis of `len` only returning `int`s. Even if we were to store `i64`s, it wouldn't matter because the underlying buffer wouldn't be able to return a sensible value in that range, assuming `int` was 32 bits and someone went above 2^31-1. This is the less breaking change, API-wise, too.